src/shogun/machine/Machine.cpp

```diff
@@ -54,6 +59,7 @@ bool CMachine::train(CFeatures* data)
 		m_labels->ensure_valid(get_name());
 	}
 
+	connect_to_signal_handler();
 	bool result = train_machine(data);
 
```

Comment:
once a machine is trained (prematurely stopped or not), shouldn't we disconnect here and reset the cancel_computation state so if one wants to re-train the same machine could still do it?